sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_client.py

```diff
@@ -74,26 +74,24 @@ def _build_auth(credential):
 def _build_connection_policy(kwargs):
     # type: (Dict[str, Any]) -> ConnectionPolicy
     # pylint: disable=protected-access
-    policy = kwargs.pop('connection_policy', None) or ConnectionPolicy()
+    policy = kwargs.pop('connection_policy', ConnectionPolicy())
 
     # Connection config
-    policy.RequestTimeout = kwargs.pop('request_timeout', None) or \
-        kwargs.pop('connection_timeout', None) or \
-        policy.RequestTimeout
-    policy.ConnectionMode = kwargs.pop('connection_mode', None) or policy.ConnectionMode
-    policy.ProxyConfiguration = kwargs.pop('proxy_config', None) or policy.ProxyConfiguration
-    policy.EnableEndpointDiscovery = kwargs.pop('enable_endpoint_discovery') \
-        if 'enable_endpoint_discovery' in kwargs.keys() else policy.EnableEndpointDiscovery
-    policy.PreferredLocations = kwargs.pop('preferred_locations', None) or policy.PreferredLocations
-    policy.UseMultipleWriteLocations = kwargs.pop('multiple_write_locations', None) or \
-        policy.UseMultipleWriteLocations
+    # `request_timeout` is supported as a legacy parameter later replaced by `connection_timeout`
+    policy.RequestTimeout = kwargs.pop('request_timeout', None) \
+                            or kwargs.pop('connection_timeout', policy.RequestTimeout)
```

Comment:
Based on the above comment, maybe we need something like:
```python
if `request_timeout` in kwargs:
    policy.RequestTimeout = kwargs['request_timeout']/1000.0
else:
    policy.RequestTimeout = kwargs.get('connection_timeout', policy.RequestTimeout)
```
Though it's also worth check what that default value in `policy.RequestTimeout` is (if anything) to make sure whether it's in seconds or milliseconds.

Reply:
Ah - looks like you've already updated the default to seconds.

Reply:
yeah the way this was being used before was as such:
<img width="412" alt="image" src="https://github.com/Azure/azure-sdk-for-python/assets/30335873/cff77fe0-48b9-4494-8cf8-9f3e262cb874">